### ISSUE TITLE
Remove --parallelism flag

### ISSUE THREAD

**Issue by skschneider** [GoogleCloudPlatform/PerfKitBenchmarker#616]
After discussing this topic within Google, the current consensus is that maintaining this flag introduces too many complexities into the PKB code when alternatives like xargs already exist.

I intend to leave this issue open for the next week in case there are any strong reasons in favor of keeping the flag. If not, the flag will be removed for the next tagged release.

**Comment by mateusz-blaszkowski** [comment]
We have a use case for parallelism flag. To give you a quick context:
we use Perfkit in a different purpose than it was originally designed for - we don't compare cloud offerings. Instead we would like to use it as a tool which helps us to stress our private clouds so that we can see where the bottlenecks are and how we should configure the cloud (in terms of hardware and software) to gain the best possible performance. 
One of the tests we perform is the oversubscription test in Openstack cloud. We heavy-load the node with multiple instances, each of them running some kind of a workload. For DB workload we use cassandra_ycsb benchmark from Perfkit. Apart from this we also spawn instances with web workload (e.g. Magento) and idle instnaces. If Perfkit had some kind of web-workload, we would use parallelism flag so that both cassandra and web-workload would be executed together in parallel. It would be nice to have both logs and results in one place (this would be treated as one run). If parallelism flag was removed, we would have to write some kind of a wrapper which would execute different PKB concurrently and also merge the results from different runs. 

Thanks,
Mateusz

**Comment by noahl** [comment]
Hi Mateusz,

For the parallelism itself, I think you can accomplish everything you want with GNU Parallel. However, I don't know about merging the results. What sort of output do you want from the combined run? Is it more complicated than appending the lists of samples generated by each benchmark?

Thanks,
Noah

**Comment by voellm** [comment]
Are you looking for results to have the same run_url or run_id?  It is
possible today to include more metadata with the --metadata flag and with
this you can link runs with a custom field.

**Comment by mateusz-blaszkowski** [comment]
Sorry for the delay. I've consulted this with other team members - it looks like we will get by without --parallelism flag. It would be nice to have this feature integrated but it's not necessary - as you pointed out running multiple PKB instances and combining the output shouldn't be a problem. Anyway, thanks for the cleanup and 'accept failures' policy - it very clearly presents the final status!

**Comment by skschneider** [comment]
Glad this is acceptable. I'll work on removing the flag later today.

**Comment by skschneider** [comment]
Removed in #633.
